mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -350,7 +345,6 @@ private class LeastSquaresAggregator(
         }
       }
       lossSum += diff * diff / 2.0
-      diffSum += diff
```

Comment:
Basically you're saying this isn't read at all, which seems correct. I don't know if it's 0? (Haven't thought about it.) The rest looks like style changes which seems OK, but why remove the `Array` type, but keep the `Long` type, when you are making the literal a long explicitly?